ExchangeSharp/API/Exchanges/Digifnex/ExchangeDigifinexAPI.cs

```diff
@@ -0,0 +1,514 @@
+            JToken token = await MakeJsonRequestAsync<JToken>("/spot/order/cancel", payload: payload, requestMethod: "POST");
+            //{
+            //  "code": 0,
+            //  "success": [
+            //    "198361cecdc65f9c8c9bb2fa68faec40",
+            //    "3fb0d98e51c18954f10d439a9cf57de0"
+            //  ],
+            //  "error": [
+            //    "78a7104e3c65cc0c5a212a53e76d0205"
+            //  ]
+            //}
+        }
+
+        #endregion
+
+        #region WebSocket APIs
+
+        protected override IWebSocket OnGetTradesWebSocket(Func<KeyValuePair<string, ExchangeTrade>, Task> callback, params string[] marketSymbols)
+        {
+            if (callback == null)
+                return null;
+            return ConnectWebSocket(string.Empty, async (_socket, msg) =>
+            {
+                // {
+                //    "method": "trades.update",
+                //    "params": 
+                //        [
+                //             true,
+                //             [
+                //                 {
+                //                 "id": 7172173,
+                //                 "time": 1523339279.761838,
+                //                 "price": "398.59",
+                //                 "amount": "0.027",
+                //                 "type": "buy"
+                //                 }
+                //             ],
+                //			 "ETH_USDT"
+                //         ],
+                //     "id": null
+                // }
+                JToken token = JToken.Parse(CryptoUtility.DecompressDeflate((new ArraySegment<byte>(msg, 2, msg.Length-2)).ToArray()).ToStringFromUTF8());
+                if (token["method"].ToStringInvariant() == "trades.update")
+                {
+                    var args = token["params"];
+                    var clean = (bool)args[0];
+                    var trades = args[1];
+                    var symbol = (string)args[2];
+
+                    var x = trades as JArray;
+                    for (int i=0; i<x.Count; i++)
+                    {
+                        var trade = x[i];
+                        var isbuy = (string)trade["type"] != "sell";
+                        var flags = default(ExchangeTradeFlags);
+                        if (isbuy)
+                            flags |= ExchangeTradeFlags.IsBuy;
+                        if (clean)
+                            flags |= ExchangeTradeFlags.IsFromSnapshot;
+                        if (i == x.Count - 1)
+                            flags |= ExchangeTradeFlags.IsLastFromSnapshot;
+
+                        await callback.Invoke(new KeyValuePair<string, ExchangeTrade>(
+                            symbol, new ExchangeTrade
+                            {
+                                Id = (string)trade["id"],
+                                Timestamp = DateTimeOffset.FromUnixTimeSeconds(0).AddSeconds((double)trade["time"]).LocalDateTime,
```

Comment:
There is a method CryptoUtility.UnixTimeStampToDateTimeSeconds(double) that can help you do this conversion but keep the timestamp in UTC.

Reply:
Oh I thought local time is more helpful in some situations

Reply:
If any end user wants local time, they can call the appropriate method on the DateTime object which will be in UTC by default and easily able to do the conversion.

Reply:
Having different timezones in different places introduces inconsistencies and may lead to inadvertent bugs in the user's code